From 3b38a22532b2461e6eba9b1aadb02e337026ddaa Mon Sep 17 00:00:00 2001
From: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Date: Fri, 15 Nov 2024 15:52:24 +0530
Subject: [PATCH] drivers: md: Handle garbage value for hwkey

To handle garbage values when the type is not mentioned and hwkey is
used, add a default type check to ensure process valid data.

Change-Id: I3d305393c800d8c3e7c3a08e90f6d12c4e36b61d
Signed-off-by: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
---
 drivers/md/dm-crypt.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/md/dm-crypt.c b/drivers/md/dm-crypt.c
index 451c7e6d103c..00433e77730e 100644
--- a/drivers/md/dm-crypt.c
+++ b/drivers/md/dm-crypt.c
@@ -3277,7 +3277,7 @@ static int crypt_ctr_cipher(struct dm_target *ti, char *cipher_in, char *key)
 }
 
 #ifdef CONFIG_BLK_INLINE_ENCRYPTION
-static int qcom_set_ice_context(struct dm_target *ti, char **argv)
+static int qcom_set_ice_context(struct dm_target *ti, u32 argc, char **argv)
 {
 	struct crypt_config *cc = ti->private;
 	uint8_t *hex_data_context = NULL, *hex_salt_context = NULL;
@@ -3309,7 +3309,7 @@ static int qcom_set_ice_context(struct dm_target *ti, char **argv)
 		return -EINVAL;
 	}
 
-	if (argv[8] != NULL && !strcmp(argv[8], "oemseed")) {
+	if (argc > 8 && argv[8] != NULL && !strcmp(argv[8], "oemseed")) {
 		seedtype = OEM_SEED_TYPE;
 	} else {
 		ret = qcom_context_ice_sec(seedtype, key_size, algo_mode,
@@ -3555,7 +3555,7 @@ static int crypt_ctr(struct dm_target *ti, unsigned int argc, char **argv)
 
 #ifdef CONFIG_BLK_INLINE_ENCRYPTION
 	if (test_bit(DM_CRYPT_INLINE_ENCRYPTION_USE_HWKEY, &cc->flags)) {
-		ret = qcom_set_ice_context(ti, argv);
+		ret = qcom_set_ice_context(ti, argc, argv);
 		if (ret < 0)
 			goto bad;
 	}
-- 
2.34.1

